spec/models/github_organization_spec.rb

```diff
@@ -1,63 +1,131 @@
 require 'rails_helper'
 
 describe GitHubOrganization do
-  let(:organization) { GitHubFactory.create_owner_classroom_org }
+  it_behaves_like 'a GitHubResource descendant with attributes'
 
   before do
     Octokit.reset!
-    @client              = oauth_client
-    @repo_name           = 'test-github-repository'
-    @github_organization = GitHubOrganization.new(@client, organization.github_id)
   end
 
-  describe '#admin?', :vcr do
-    it 'verifies if the user is an admin of the organization' do
-      user         = organization.users.first
-      github_admin = GitHubUser.new(user.github_client, user.uid)
-      expect(@github_organization.admin?(github_admin.login)).to eql(true)
-    end
-  end
+  context 'with GitHub Organization', :vcr do
+    let(:organization)        { GitHubFactory.create_owner_classroom_org }
+    let(:github_organization) { organization.github_organization }
+    let(:repo_name)           { 'test-github-repository' }
+
+    subject { organization.github_organization }
 
-  describe '#create_repository', :vcr do
-    after do
-      @client.delete_repository("#{organization.title}/#{@repo_name}")
+    describe '#add_membership' do
     end
 
-    it 'successfully creates a GitHub Repository for the Organization' do
-      @github_organization.create_repository(@repo_name, private: true)
-      expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
+    context 'with GitHub repository' do
+      before(:each) do
```

Comment:
do you need these before and after blocks in both cases? Seems like you're trying to delete a repository twice.